lib/types.nix

```
@@ -137,47 +137,60 @@ rec {
       }";
     };
 
-  pluginLuaConfig = types.submodule (
-    { config, ... }:
-    let
-      inherit (builtins) toString;
-      inherit (lib.nixvim.utils) mkBeforeSection mkAfterSection;
-    in
-    {
-      options = {
-        pre = lib.mkOption {
-          type = with types; nullOr lines;
-          default = null;
-          description = ''
-            Lua code inserted at the start of the plugin's configuration.
-            This is the same as using `lib.nixvim.utils.mkBeforeSection` when defining `content`.
-          '';
-        };
-        post = lib.mkOption {
-          type = with types; nullOr lines;
-          default = null;
-          description = ''
-            Lua code inserted at the end of the plugin's configuration.
-            This is the same as using `lib.nixvim.utils.mkAfterSection` when defining `content`.
-          '';
-        };
-        content = lib.mkOption {
-          type = types.lines;
-          default = "";
-          description = ''
-            Configuration of the plugin.
+  pluginLuaConfig =
+    { hasContent }:
+    types.submodule (
+      { config, ... }:
+      let
+        inherit (builtins) toString;
+        inherit (lib.nixvim.utils) mkBeforeSection mkAfterSection;
+      in
+      {
+        options =
+          {
+            pre = lib.mkOption {
+              type = with types; nullOr lines;
+              default = null;
+              description =
+                ''
+                  Lua code inserted at the start of the plugin's configuration.
+                ''
+                + lib.optionalString hasContent ''
+                  This is the same as using `lib.nixvim.utils.mkBeforeSection` when defining `content`.
+                '';
+            };
+            post = lib.mkOption {
+              type = with types; nullOr lines;
+              default = null;
+              description =
+                ''
+                  Lua code inserted at the end of the plugin's configuration.
+                ''
+                + lib.optionalString hasContent ''
+                  This is the same as using `lib.nixvim.utils.mkAfterSection` when defining `content`.
+                '';
+            };
+          }
+          // (lib.optionalAttrs hasContent) {
```

```suggestion
          // lib.optionalAttrs hasContent {
```

I don't think this would cause eval errors, but it seems like a typo left over from some previous code?